scripts/test-tls13-certificate-request.py

```diff
@@ -256,6 +263,59 @@ def main():
 
     conversations["check sigalgs in cert request"] = conversation
 
+    # verify the sent extensions
+    conversation = Connect(hostname, port)
+    node = conversation
+    ciphers = [CipherSuite.TLS_AES_128_GCM_SHA256,
+               CipherSuite.TLS_EMPTY_RENEGOTIATION_INFO_SCSV]
+    ext = {}
+    groups = [GroupName.secp256r1]
+    key_shares = []
+    for group in groups:
+        key_shares.append(key_share_gen(group))
+    ext[ExtensionType.key_share] = \
+        ClientKeyShareExtension().create(key_shares)
+    ext[ExtensionType.supported_versions] = \
+        SupportedVersionsExtension().create([(3, 4), (3, 3)])
```

Comment:
The (3,4) and (3,3) versions are pretty ubiquitous. it seems like having a tls1_3version and tls1_2version define might make this more readable (color the spelling however you want to fit tlsfuzzer style). Not necessary in this patch because I see it everywhere, but it seems like a reasonable readability improvement (I looked at a fair amount of the patch before I realized this was the major and minor tls version number).

Reply:
good point, filed https://github.com/tomato42/tlslite-ng/issues/370 to fix it in the future